src/core/auth/qgsauthmanager.h

```diff
@@ -248,6 +249,21 @@ class CORE_EXPORT QgsAuthManager : public QObject
      */
     bool resetMasterPassword( const QString &newpass, const QString &oldpass, bool keepbackup, QString *backuppath SIP_INOUT = nullptr );
 
+    /**
+     * Reset the master password to a new one, then re-encrypt all previous
+     * configs in a new database file, optionally backup current database.
```

Comment:
unless this is specific to the sqlite storage better to keep it generic
```suggestion
     * configs in a new credentials storage optionally backup the current storage if the storage type supports backups.
```